os/kernel/task/task_dumptasks.c

```diff
@@ -0,0 +1,134 @@
+ * Private Type Declarations
+ ****************************************************************************/
+
+/****************************************************************************
+ * Global Variables
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Variables
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: up_show_tcbinfo
+ ****************************************************************************/
+
+void task_show_tcbinfo(struct tcb_s *tcb)
+{
+	lldbg("State       : %u\n", tcb->task_state);
+	lldbg("Flags       : %u\n", tcb->flags);
+	lldbg("Lock count  : %u\n", tcb->lockcount);
+#if CONFIG_RR_INTERVAL > 0
+	lldbg("Timeslice   : %d\n", tcb->timeslice);
+#endif
+	lldbg("Waitdog     : %p\n", tcb->waitdog);
+	lldbg("WaitSem     : %p\n", tcb->waitsem);
+#ifndef CONFIG_DISABLE_MQUEUE
+	lldbg("MsgwaitQ    : %p\n", tcb->msgwaitq);
+#endif
+#ifndef CONFIG_DISABLE_SIGNALS
+	lldbg("Sigdeliver  : %p\n", tcb->xcp.sigdeliver);
+#endif
+#ifdef CONFIG_LIB_SYSCALL
+	lldbg("Nsyscalls   : %u\n", tcb->xcp.nsyscalls);
+	lldbg("Syscall     : %p\n", tcb->xcp.syscall);
+#endif
+}
+
+/****************************************************************************
+ * Name: task_taskdump
+ ****************************************************************************/
+
+#ifdef CONFIG_STACK_COLORATION
+static void task_taskdump(FAR struct tcb_s *tcb, FAR void *arg)
+{
+	size_t used_stack_size;
+
+	used_stack_size = up_check_tcbstack(tcb);
+
+	/* Dump interesting properties of this task */
+
+#if CONFIG_TASK_NAME_SIZE > 0
+	lldbg("%*s | %5d | %4d | %7lu / %7lu | %16p | %8p\n", CONFIG_TASK_NAME_SIZE,
+			tcb->name, tcb->pid, tcb->sched_priority,  
+			(unsigned long)used_stack_size, (unsigned long)tcb->adj_stack_size, tcb->stack_alloc_ptr, tcb);
+#else
+	lldbg("%5d | %4d | %7lu / %7lu | %16p | %8p\n",
+			tcb->pid, tcb->sched_priority, tcb, (unsigned long)used_stack_size,
+			(unsigned long)tcb->adj_stack_size, tcb->stack_alloc_ptr, tcb);
+#endif
+
+	if (used_stack_size == tcb->adj_stack_size) {
+		lldbg("  !!! PID (%d) STACK OVERFLOW !!! \n", tcb->pid);
+	}
+
+}
+#endif
+
+/****************************************************************************
+ * Name: up_showtasks
+ ****************************************************************************/
+
+#ifdef CONFIG_STACK_COLORATION
+void task_show_tasks(void)
```

Comment:
Naming looks not perfectly showing what function does.
How about `task_show_alivetask_list`?